funcx_endpoint/funcx_endpoint/executors/high_throughput/worker_map.py

```diff
@@ -75,6 +98,11 @@ def remove_worker(self, worker_id):
         self.total_worker_type_counts["unused"] += 1
         self.ready_worker_type_counts["unused"] += 1
 
+        # Mark the accelerator as available, if provided
+        if worker_id in self.assigned_accelerators:
+            device = self.assigned_accelerators.pop(worker_id)
+            self.available_accelerators.put(device)
+
```

Comment:
I'm concerned with how this interacts with `TASK_CANCEL`.

`remove_worker` will return an accelerator to the pool of assignable accelerators, however in the case of `TASK_CANCEL`, this is *before* a task has been cancelled. That means an accelerator could be assigned to a new worker while an old worker is still using it.

Tagging @yadudoc here who has more task cancel behaviour in his head than me - maybe the right thing is for `remove_worker` to be called *after* the worker process has been killed, in the `TASK_CANCEL` code.

Reply:
I'll defer to Yadu on this one too. I don't have a good grasp on when workers die and only tested this with worker timeout.

Reply:
@benclifford Looks like you're right. In the `task_cancel` path, we call `remove_worker` before we terminate the worker. And that can take up to 1s. To avoid a race condition we ought to move the `remove_worker` call [here](https://github.com/funcx-faas/funcX/blob/main/funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_manager.py#L365) to [here](https://github.com/funcx-faas/funcX/blob/main/funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_manager.py#L382).

At least that's what I think needs to happen here.

Reply:
@yadudoc yes i think that's the right change. remove_worker even says: `        Should already be KILLed by this point.` in its docstring

Reply:
Should I make that change, or do you want it part of another PR?

Reply:
make it here